src/main/scala/org/constellation/util/MajorityStateChooser.scala

```diff
@@ -3,55 +3,100 @@ package org.constellation.util
 import cats.data.OptionT
 import cats.effect.Concurrent
 import cats.implicits._
-import io.chrisdavenport.log4cats.Logger
 import io.chrisdavenport.log4cats.slf4j.Slf4jLogger
-import org.constellation.primitives.Schema._
+import org.constellation.domain.redownload.ReDownloadPlan
 import org.constellation.schema.Id
 import org.constellation.storage.RecentSnapshot
 
-import scala.util.{Random, Try}
-
-class MajorityStateChooser[F[_]: Concurrent] {
+import scala.util.Try
 
+object MajorityStateChooser {
```

Comment:
Why have you removed the class and moved everything into an object? That way you're not able to mock this implementation and code become un-testable.

Reply:
True, but they're all pure functions so we don't need mocks in MajorityStateChooserTest. I made it an object because they're all pure functions, and MajorityStateChooser doesn't need F